src/snowflake/cli/plugins/nativeapp/run_processor.py

```diff
@@ -141,6 +143,38 @@ def _create_dev_app(self, diff: DiffResult) -> None:
             except ProgrammingError as err:
                 generic_sql_error_handler(err)
 
+            self._execute_post_deploy_hooks()
+
+    def _execute_sql_script(self, sql_script_path):
+        """
+        Executing the SQL script in the provided file path after expanding template variables.
+        "use warehouse" and "use database" will be executed first if they are set in definition file or in the current connection.
+        """
+        with open(sql_script_path) as f:
+            sql_script = f.read()
+            try:
+                if self.application_warehouse:
+                    self._execute_query(f"use warehouse {self.application_warehouse}")
+                if self._conn.database:
+                    self._execute_query(f"use database {self._conn.database}")
+                sql_script = snowflake_sql_jinja_render(content=sql_script)
+                self._execute_queries(sql_script)
+            except ProgrammingError as err:
+                generic_sql_error_handler(err)
+
+    def _execute_post_deploy_hooks(self):
+        post_deploy_script_hooks = self.app_post_deploy_hooks
+        if post_deploy_script_hooks:
+            with cc.phase("Executing application post-deploy actions..."):
+                for hook in post_deploy_script_hooks:
+                    if hook.sql_script:
+                        cc.step(f"Executing SQL script: {hook.sql_script}")
+                        self._execute_sql_script(hook.sql_script)
+                    else:
+                        raise ValueError(
+                            f"Unsupported application post-deploy hook type: {hook}"
+                        )
```

Comment:
Consider validating upfront so we don't spend time/credits executing something just to have the whole flow fail later on. Better CX that way.

Reply:
This validation is in addition to the schema validation which catches non-supported hook types on command execution:
```
During evaluation of ApplicationPostDeployHook in project definition following errors were encountered:
Extra inputs are not permitted. You provided field '('non_existing_hook',)' with value 'scripts/post_install.sql' that is not supported in given version.
```

Reply:
So this is dead code?

Reply:
In a way. It's an additional protection layer: someone could change the schema without updating the logic. Something could also change the definition object dynamically.

Maybe it's personal preference, but I'm not comfortable with having an `if` condition and assume the `else` flow is not reachable.